common/app/model/CardStylePicker.scala

```diff
@@ -11,22 +11,33 @@ object CardStylePicker {
 
   def apply(content: CapiContent): CardStyle = {
     val tags = content.tags.map(_.id).toSeq
-    extractCampaigns(tags) match {
-      case Nil                                                                         => CardStyle(content, TrailMetaData.empty)
-      case campaign :: Nil if campaign.id.toString.toLowerCase() == "specialreportalt" => SpecialReportAlt
-      case _                                                                           => SpecialReport
+    val campaigns = extractCampaigns(tags)
+    campaigns match {
+      case Nil => CardStyle(content, TrailMetaData.empty)
+      case _   => getCardStyleForReport(campaigns)
     }
   }
 
   def apply(content: FaciaContent): CardStyle = {
     val tags = FaciaContentUtils.tags(content).map(_.id)
-    extractCampaigns(tags) match {
-      case Nil                                                                         => FaciaContentUtils.cardStyle(content)
-      case campaign :: Nil if campaign.id.toString.toLowerCase() == "specialreportalt" => SpecialReportAlt
-      case _                                                                           => SpecialReport
+    val campaigns = extractCampaigns(tags)
+    campaigns match {
+      case Nil => FaciaContentUtils.cardStyle(content)
+      case _   => getCardStyleForReport(campaigns)
     }
   }
 
+  def getCardStyleForReport(campaigns: List[Campaign]): CardStyle = {
+    if (containsSpecialReportAltCampaign(campaigns)) SpecialReportAlt else SpecialReport
+  }
+
+  private def containsSpecialReportAltCampaign(campaigns: List[Campaign]): Boolean = {
+    val specialReportAltCampaigns = campaigns.filter(campaign =>
+      campaign.fields.asInstanceOf[ReportFields].campaignId.toLowerCase() == "specialreportalt",
```

Comment:
Would you mind spelling out why we use `.asInstanceOf` here? I'm not super familiar with it but it looks like it could lead to runtime errors if the fields weren't `ReportFields`? Is that the behaviour we'd want here if other kinds of fields were passed?

Reply:
Thanks for the question @bryophyta! @JamieB-gu and I have refactored the code so that it doesn't use `asInstanceOf` anymore and hopefully makes the code clearer! But to answer your initial questions:  

At this point we already had a list of campaigns of type `report` (i.e. Special Reports, like the "Pandora Papers" or "Swiss Secrets", or SpecialReportAlt, there's only the one we are preparing). These were being filtered in the [`extractCampaigns` method (https://github.com/guardian/frontend/blob/bc5d6427ef3e9a4f1c0026fac240689659c3ffa2/common/app/model/CardStylePicker.scala#L30-L33), so we already had eliminated the case for a runtime error. However, this wasn't clear at all! Hopefully, this new refactoring makes it clearer.

On the _why_ we had to somehow make sure we are checking the `campaignId` of `ReportFields`, the https://targeting.gutools.co.uk/ has a few types of campaigns, e.g. report, epic, callout etc. [`Campaign`](https://github.com/guardian/targeting-client/blob/main/src/main/scala/com/gu/targeting/client/Campaign.scala#L14-L22) varies on [`Fields`](https://github.com/guardian/targeting-client/blob/main/src/main/scala/com/gu/targeting/client/Fields.scala#L11) which is a trait with various implementations. A`SpecialReportAlt` is a `Campaign` that has `ReportFields`, so this method was using `asInstanceOf` in order to access its `campaignId`. Hopefully the [new version of the code](https://github.com/guardian/frontend/pull/25991/commits/b3a2b35c2d564860df3d5891a1551b96aa9cfab1#diff-5c16a53fac591563d934dcd448f2c7a16160e78d4d3e6faa501e56ce6a9b15dfR43) is clearer but let me know if not!